pyccel/ast/utilities.py

```diff
@@ -177,3 +186,398 @@ def split_positional_keyword_arguments(*args):
+                # and the loop is not used to modify one of the variable dependencies
+                if save_spot[-1].length == shape[j] and not any(u in save_spot[-1].modified_vars for u in dependencies):
+                    save_spot[-1].modified_vars.update(lhs_vars)
+                    save_spot = save_spot[-1].body
+                    j+=1
+                else:
+                    break
+
+            for k in range(j,new_level):
+                # Create new loops until we have the neccesary depth
+                save_spot.append(LoopCollection([], shape[k], set(lhs_vars)))
+                save_spot = save_spot[-1].body
+
+            # Save results
+            save_spot.append(line)
+            current_level = new_level
+        else:
+            # Save line in top level (no for loop)
+            result.append(line)
+            current_level = 0
+    return result
+
+#==============================================================================
+
+def insert_fors(blocks, indices, level = 0):
+    """
+    Run through the output of collect_loops and create For loops of the
+    requested sizes
+
+    Parameters
+    ==========
+    block   : list of LoopCollection
+            The result of a call to collect_loops
+    indices : list
+            The index variables
+    level   : int
+            The index of the index variable used in the outermost loop
+    Results
+    =======
+    block : list of PyccelAstNodes
+            The modified expression
+    """
+    if all(not isinstance(b, LoopCollection) for b in blocks.body):
+        body = blocks.body
+    else:
+        body = [insert_fors(b, indices, level+1) if isinstance(b, LoopCollection) else [b] \
+                for b in blocks.body]
+        body = [bi for b in body for bi in b]
+    if blocks.length == 1:
+        return body
+    else:
+        return [For(indices[level], PythonRange(0,blocks.length), body)]
+
+#==============================================================================
+def expand_tuple_assignments(block):
+    """
+    Simplify expressions in a CodeBlock by unravelling tuple assignments into multiple lines
+
+    Parameters
+    ==========
+    block      : CodeBlock
+                The expression to be modified
+
+    Examples
+    --------
+    >>> from pyccel.ast.core import Variable, Assign
+    >>> from pyccel.ast.operators import PyccelAdd
+    >>> from pyccel.ast.utilities import expand_to_loops
+    >>> a = Variable('int', 'a', shape=(4,), rank=1)
+    >>> b = Variable('int', 'b', shape=(4,), rank=1)
+    >>> c = Variable('int', 'c', shape=(4,), rank=1)
+    >>> i = Variable('int', 'i', shape=())
+    >>> d = PyccelAdd(a,b)
+    >>> expr = [Assign(c,d)]
+    >>> expand_to_loops(expr, language_has_vectors = False)
+    [For(i_0, PythonRange(0, LiteralInteger(4), LiteralInteger(1)), CodeBlock([IndexedElement(c, i_0) := PyccelAdd(IndexedElement(a, i_0), IndexedElement(b, i_0))]), [])]
```

Comment:
This example is for function `expand_to_loops`, isn't it?

Reply:
Ah you are right, I meant to go back and fix this but I forgot. Fixed in 3b78db5